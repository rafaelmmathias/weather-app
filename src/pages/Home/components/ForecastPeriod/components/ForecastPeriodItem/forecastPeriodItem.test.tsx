import { render, screen } from "@testing-library/react";
import { ForecastPeriodItem, ForecastPeriodItemProps } from ".";
import renderer from "react-test-renderer";

let mockedProps: ForecastPeriodItemProps = {
  item: {
    name: "Today",
    value: [
      {
        detailedForecast: "Partly sunny. High near 98",
        endTime: "2022-04-26T18:00:00-07:00",
        icon: "fakeIcon",
        isDaytime: true,
        name: "Today",
        shortForecast: "Partly Sunny",
        startTime: "2022-04-26T12:00:00-07:00",
        temperature: 98,
        temperatureUnit: "F",
        windDirection: "SSW",
        windSpeed: "5 to 10 mph",
      },
    ],
  },
};

test("expected to render the forecast title and detailed text", () => {
  render(<ForecastPeriodItem {...mockedProps} />);

  const titleElement = screen.getByText(/Today/i);
  const detailedForecast = screen.getByText(/Partly sunny. High near 98/i);

  expect(titleElement).toBeInTheDocument();
  expect(detailedForecast).toBeInTheDocument();
});

test("expected to render correctly when have just 1 item", () => {
  const tree = renderer
    .create(<ForecastPeriodItem {...mockedProps} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("expected to render correctly when have more than 1", () => {
  let _mockedPropss: ForecastPeriodItemProps = JSON.parse(
    JSON.stringify(mockedProps)
  );

  _mockedPropss.item.value.push({
    detailedForecast: "detailed 2",
    endTime: "2022-04-26T18:00:00-07:00",
    icon: "second fakeIcon",
    isDaytime: true,
    name: "Tonight",
    shortForecast: "Partly Sunny",
    startTime: "2022-04-26T12:00:00-07:00",
    temperature: 78,
    temperatureUnit: "F",
    windDirection: "SSW",
    windSpeed: "5 to 10 mph",
  });

  const tree = renderer
    .create(<ForecastPeriodItem {..._mockedPropss} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
