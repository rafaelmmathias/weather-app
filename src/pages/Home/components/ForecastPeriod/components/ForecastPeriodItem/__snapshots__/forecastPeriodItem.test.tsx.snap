// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expected to render correctly when have just 1 item 1`] = `
<div
  className="sc-bczRLJ dLEboF sc-gsnTZi gnXpTX"
>
  <div
    className="sc-bczRLJ VvvOl"
    display="flex"
  >
    <div
      className="sc-bczRLJ cBQUGH sc-gKXOVf ictodb"
      data-testid="forecast-period-item"
    >
      <div
        className="sc-bczRLJ bRqHoP"
        display="flex"
      >
        <div
          className="sc-bczRLJ yfIxC"
          display="flex"
        >
          <span
            className="sc-hKMtZM bdEyhf"
            color="#fff"
            data-testid="forecast-card-item-name"
            fontSize={20}
          >
            Today
          </span>
          <span
            className="sc-hKMtZM iauWbn"
            color="#fff"
          >
            Partly Sunny
          </span>
        </div>
        <div
          className="sc-bczRLJ bFjlUq"
          display="flex"
          width={90}
        >
          <span
            className="sc-hKMtZM jfvJoT"
            color="yellow"
            data-testid="forecast-card-item-temperature"
            fontSize={25}
          >
            98° F
          </span>
        </div>
      </div>
      <div
        className="sc-bczRLJ jKunKD"
        overflow="auto"
      >
        <p
          className="sc-dkzDqf dMhlLK"
          color="#fff"
        >
          Partly sunny. High near 98
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`expected to render correctly when have more than 1 1`] = `
<div
  className="sc-bczRLJ dLEboF sc-gsnTZi gnXpTX"
>
  <div
    className="sc-bczRLJ VvvOl"
    display="flex"
  >
    <div
      className="sc-bczRLJ cBQUGH sc-gKXOVf ictodb"
      data-testid="forecast-period-item"
    >
      <div
        className="sc-bczRLJ bRqHoP"
        display="flex"
      >
        <div
          className="sc-bczRLJ yfIxC"
          display="flex"
        >
          <span
            className="sc-hKMtZM bdEyhf"
            color="#fff"
            data-testid="forecast-card-item-name"
            fontSize={20}
          >
            Today
          </span>
          <span
            className="sc-hKMtZM iauWbn"
            color="#fff"
          >
            Partly Sunny
          </span>
        </div>
        <div
          className="sc-bczRLJ bFjlUq"
          display="flex"
          width={90}
        >
          <span
            className="sc-hKMtZM jfvJoT"
            color="yellow"
            data-testid="forecast-card-item-temperature"
            fontSize={25}
          >
            98° F
          </span>
        </div>
      </div>
      <div
        className="sc-bczRLJ jKunKD"
        overflow="auto"
      >
        <p
          className="sc-dkzDqf dMhlLK"
          color="#fff"
        >
          Partly sunny. High near 98
        </p>
      </div>
    </div>
    <div
      className="sc-bczRLJ dNZAiX sc-gKXOVf ictodb"
      data-testid="forecast-period-item"
    >
      <div
        className="sc-bczRLJ bRqHoP"
        display="flex"
      >
        <div
          className="sc-bczRLJ yfIxC"
          display="flex"
        >
          <span
            className="sc-hKMtZM bdEyhf"
            color="#fff"
            data-testid="forecast-card-item-name"
            fontSize={20}
          >
            Tonight
          </span>
          <span
            className="sc-hKMtZM iauWbn"
            color="#fff"
          >
            Partly Sunny
          </span>
        </div>
        <div
          className="sc-bczRLJ bFjlUq"
          display="flex"
          width={90}
        >
          <span
            className="sc-hKMtZM jfvJoT"
            color="yellow"
            data-testid="forecast-card-item-temperature"
            fontSize={25}
          >
            78° F
          </span>
        </div>
      </div>
      <div
        className="sc-bczRLJ jKunKD"
        overflow="auto"
      >
        <p
          className="sc-dkzDqf dMhlLK"
          color="#fff"
        >
          detailed 2
        </p>
      </div>
    </div>
  </div>
</div>
`;
